model/metrics/clr-metrics.yaml

```diff
@@ -0,0 +1,167 @@
+groups:
+  - id: metric.clr.gc.collections.count
+    type: metric
+    metric_name: clr.gc.collections.count
+    brief: "Number of garbage collections that have occurred since the process has started."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.objects.size
+    type: metric
+    metric_name: clr.gc.objects.size
+    brief: "The number of bytes currently allocated on the managed GC heap. Fragmentation and other GC committed memory pools are excluded."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.allocations.size
+    type: metric
+    metric_name: clr.gc.allocations.size
+    brief: >
+      The number of bytes allocated on the managed GC heap since the process has
+      started. The returned value does not include any native allocations.
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.committed_memory.size
+    type: metric
+    metric_name: clr.gc.committed_memory.size
```

Comment:
There are other forms of memory that .NET processes use that isn't the gc so using "gc" here does feel more precise and allows us to add metrics for other forms of memory usage in the future. For example loading assemblies allocates memory for modules, types, and jitted code. Even some .NET objects are not allocated on the gc heap: https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#non-gc-heap

Reply:
> There are other forms of memory that .NET processes use that isn't the gc so using "gc" here does feel more precise and allows us to add metrics for other forms of memory usage in the future. For example loading assemblies allocates memory for modules, types, and jitted code. Even some .NET objects are not allocated on the gc heap: https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#non-gc-heap

+1, similar to https://github.com/open-telemetry/semantic-conventions/pull/1035#discussion_r1621737253

Reply:
Something feels off (to me) about reporting memory usage in the `gc` namespace since the memory concept is wider than `gc`. 

It's not a strong opinion, but I'd prefer to have `dotnet.memory.heap` vs `dotnet.gc.memory`. 
Java uses a different approach by adding memory type attribute `jvm.memory.type = heap | non_heap` do distinguish those.

Reply:
If we had a top-level committed memory metric a few things that seem awkward to me about that approach:
1. It gets our runtime metrics into the business of process metrics to report the committed memory used by everything else that isn't the GC.
2. In the future we might want to expose other pools of committed memory and there is no guarantee those pools are all mutually exclusive subsets of the memory. As a current example GC generation sizes are neither a subset nor a super-set of committed memory. Each generation may have part of its memory in the committed state and part of its memory in the reserved state.
3. We have a bunch of other GC memory metrics so unless we are planning to rename all of them to have a new prefix it seems weird to have them in separate namespace prefixes.

Reply:
let's step back from the naming for a second and think about non-advanced (in .NET) user. 

I do see what my .NET process consumes on the OS level from some generic external instrumentation.
Now I see a `dotnet.gc.heap` metric which is documented fairly well, but there is a gap between the sum of dimensions and what I see on the OS level - what is it? What are the metrics that report it? 
It seems there will be 
- gc heap
- non-gc heap
- native? 
- what else?

If each of them is reported as a separate metric, I need to either know all of them or look into the docs. 

Can we make this journey easy? What are the memory pools? Do we have an exhaustive list of metrics that describe all of them? 

If we can't make it easy, are we ready to document it (in a format of semconv, not as a book)?

This is the same point I'm raising in 
https://github.com/open-telemetry/semantic-conventions/pull/1035#discussion_r1624901717

Reply:
I feel the fact that there is a gc heap and non-gc heap makes my point even stronger:
How is  having `dotnet.gc.heap` and `dotnet.non_gc.heap` better than `dotnet.heap` metric with `gen-0`, `gen-1`, `loh`, `non-heap` dimensions?

You can always add new values to the `type` dimension in the same way you can add more metrics.
But users see what's there without prior knowledge of heap/non heap types.

Move one step further and you have `dotnet.memory.used(?)` and users can learn .net memory management from metric dimensions.

Reply:
Alright I think I follow better what you are after now and also sounds like I may have had the wrong understanding about the appropriateness of runtimes reporting process-level metrics. I'm waiting to get confirmation on the [cpu discussion](https://github.com/open-telemetry/semantic-conventions/pull/1035#discussion_r1630687882) above to make sure I don't create further confusion before I continue working through this one with you :)

Reply:
consolidating memory discussions in https://github.com/open-telemetry/semantic-conventions/pull/1035/files#r1649489585